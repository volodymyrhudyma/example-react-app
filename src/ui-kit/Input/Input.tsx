import React, { ChangeEvent } from 'react';
import { Wrapper, StyledInput, IconWrapper } from './styled/Input.styled';
import Icon from '../Icon/Icon';
import icons from 'src/constants/icons';

interface IInputProps {
  type: string;
  placeholder: string;
  value: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

const Input = (props: IInputProps) => (
  <Wrapper>
    <IconWrapper>
      <Icon src={icons.search} width={20} height={20} />
    </IconWrapper>
    <StyledInput {...props} />
  </Wrapper>
);

Input.defaultProps = {
  type: 'text',
};

export default Input;
