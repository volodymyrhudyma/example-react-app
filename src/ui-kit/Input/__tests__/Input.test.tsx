import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

describe('Input component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Input placeholder="Placeholder" value="" onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
