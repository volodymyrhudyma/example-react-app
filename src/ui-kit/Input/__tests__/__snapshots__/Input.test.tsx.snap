// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly 1`] = `
<styled.div>
  <styled.div>
    <Icon
      alt="Icon"
      height={20}
      src="/icons/search.png"
      width={20}
    />
  </styled.div>
  <styled.input
    onChange={[MockFunction]}
    placeholder="Placeholder"
    type="text"
    value=""
  />
</styled.div>
`;
